### PR TITLE
Improve structure, add eslint, pre commit lint check

### PR DESCRIPTION
**Pull request by RobMin**

### CODE REVIEW

**Comment by RobMin** [comment]
[Beautify](https://trello.com/c/K5AeVzRl/4-beautify)

File: .gitignore [hunk 1]
```diff
@@ -1,108 +1 @@
-# Logs
-logs
-*.log
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
-lerna-debug.log*
-
-# Parcel
-parcel_cache/
```
Comment (GagMirz) [inline .gitignore:10]: Why have you removed parcel cache ignorance

File: src/game-object/utils/event.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+interface IHandler {
+  method: () => void;
+  id: string;
+}
+
+export class Events<Trigger extends string | number> {
+  private events: { [ trigger in Trigger ]?: IHandler[] } = {};
+
+  public addEvent(trigger: Trigger, id: IHandler['id'], method: IHandler['method']): void {
```
Comment (GagMirz) [inline src/game-object/utils/event.ts:9]: Maybe addHandler is a better idea as we are adding a handler to a trigger

File: src/game-object/utils/event.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    if (!this.events[trigger]) {
+      this.events[trigger] = [];
+    }
+
+    this.events[trigger].push({ id, method });
```
Comment (GagMirz) [inline src/game-object/utils/event.ts:14]: Add checking, if there is a handler under the provided id throw exception

**Review by GagMirz — COMMENTED** [review]
Check comments

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+node_modules
+
+# Parcel
+parcel_cache/
```
Comment (GagMirz) [inline .eslintignore:4]: forgot to mention parcel is also generating dist/ folder on build and run, ignore it too

**Review by GagMirz — APPROVED** [review]
GOOD job

**Pull request merged**